ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -639,11 +649,14 @@ func (r *RayClusterReconciler) reconcilePods(ctx context.Context, instance *rayv
 		logger.Info("reconcilePods", "head Pod", headPod.Name, "shouldDelete", shouldDelete, "reason", reason)
 		if shouldDelete {
 			if err := r.Delete(ctx, &headPod); err != nil {
+				r.Recorder.Eventf(instance, corev1.EventTypeWarning, FailedRequest,
+					"Failed deleting head Pod %s/%s; Pod status: %s; Pod restart policy: %s; Ray container terminated status: %v",
+					headPod.Namespace, headPod.Name, headPod.Status.Phase, headPod.Spec.RestartPolicy, getRayContainerStateTerminated(headPod))
 				return errstd.Join(utils.ErrFailedDeleteHeadPod, err)
 			}
-			r.Recorder.Eventf(instance, corev1.EventTypeNormal, "Deleted",
-				"Deleted head Pod %s; Pod status: %s; Pod restart policy: %s; Ray container terminated status: %v",
-				headPod.Name, headPod.Status.Phase, headPod.Spec.RestartPolicy, getRayContainerStateTerminated(headPod))
+			r.Recorder.Eventf(instance, corev1.EventTypeNormal, DeleteRequestSuccess,
+				"Deleted head Pod %s/%s; Pod status: %s; Pod restart policy: %s; Ray container terminated status: %v",
+				headPod.Namespace, headPod.Name, headPod.Status.Phase, headPod.Spec.RestartPolicy, getRayContainerStateTerminated(headPod))
```

Comment:
I wanted to ask if the same event info can be surfaced here for the failure event as well? Should we include the error in here as well?

Reply:
I think it is better to include an error message whenever possible, for instance: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/kubelet.go#L2025

But do we have an error here? Line 657 seems to be a successful case.